plugins/processors/scale/scale_test.go

```diff
@@ -0,0 +1,232 @@
+					map[string]interface{}{
+						"test1": int64(0),
+						"test2": uint64(1),
+						"test3": int64(1),
+					}, time.Unix(0, 0)),
+			},
+			expected: []telegraf.Metric{
+				testutil.MustMetric("Name1", map[string]string{},
+					map[string]interface{}{
+						"test1": float64(50),
+						"test2": float64(100),
+						"test3": int64(1),
+					}, time.Unix(0, 0)),
+			},
+		},
+		{
+			name: "Out of range tests",
+			scale: &Scale{
+				Scalings: []Scaling{
+					{
+						InMin:  -1,
+						InMax:  1,
+						OutMin: 0,
+						OutMax: 100,
+						Fields: []string{"test1", "test2"},
+					},
+				},
+			},
+			inputs: []telegraf.Metric{
+				testutil.MustMetric("Name1", map[string]string{},
+					map[string]interface{}{
+						"test1": int64(-2),
+						"test2": uint64(2),
+					}, time.Unix(0, 0)),
+			},
+			expected: []telegraf.Metric{
+				testutil.MustMetric("Name1", map[string]string{},
+					map[string]interface{}{
+						"test1": float64(-50),
+						"test2": float64(150),
+					}, time.Unix(0, 0)),
+			},
+		},
+		{
+			name: "Missing field Fileds",
+			scale: &Scale{
+				Scalings: []Scaling{
+					{
+						InMin:  -1,
+						InMax:  1,
+						OutMin: 0,
+						OutMax: 100,
+						Fields: []string{"test1", "test2"},
+					},
+				},
+			},
+			inputs: []telegraf.Metric{
+				testutil.MustMetric("Name1", map[string]string{},
+					map[string]interface{}{
+						"test1": int64(0),
+					}, time.Unix(0, 0)),
+			},
+			expected: []telegraf.Metric{
+				testutil.MustMetric("Name1", map[string]string{},
+					map[string]interface{}{
+						"test1": float64(50),
+					}, time.Unix(0, 0)),
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.scale.Log = testutil.Logger{}
+
+			require.NoError(t, tt.scale.Init())
+			actual := tt.scale.Apply(tt.inputs...)
+
+			testutil.RequireMetricsEqual(t, tt.expected, actual)
+		})
+	}
+}
+
+func TestErrorCases(t *testing.T) {
+	tests := []struct {
+		name             string
+		scale            *Scale
+		expectedErrorMsg string
+	}{
+		{
+			name: "Same input range values",
+			scale: &Scale{
+				Scalings: []Scaling{
+					{
+						InMin:  1,
+						InMax:  1,
+						OutMin: 0,
+						OutMax: 100,
+						Fields: []string{"test"},
+					},
+				},
+			},
+			expectedErrorMsg: "input minimum and maximum are equal for fields test",
+		},
+		{
+			name: "Same input range values",
+			scale: &Scale{
+				Scalings: []Scaling{
+					{
+						InMin:  0,
+						InMax:  1,
+						OutMin: 100,
+						OutMax: 100,
+						Fields: []string{"test"},
+					},
+				},
+			},
+			expectedErrorMsg: "output minimum and maximum are equal for fields test",
+		},
+		{
+			name:             "No scalings",
+			scale:            &Scale{Log: testutil.Logger{}},
+			expectedErrorMsg: "no valid scalings defined",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.scale.Log = testutil.Logger{}
+
+			require.Error(t, tt.scale.Init(), tt.expectedErrorMsg)
```

Comment:
```suggestion
			require.ErrorContains(t, tt.scale.Init(), tt.expectedErrorMsg)
```
As now the message isn't checked.